deployment/BUILD

```diff
@@ -16,6 +16,16 @@ k8s_object(
     template = ":buildbuddy-app.dev.yaml",
 )
 
+k8s_object(
+    name = "buildbuddy_dev_app_oci",
+    cluster = "gke_flame-build_us-west1-a_bb-dev-cluster",
+    oci_images = {
+        "gcr.io/flame-build/buildbuddy-app-dev:server_image": "//server:oci_image",
+    },
+    tags = ["manual"],
+    template = ":buildbuddy-app.dev.yaml",
+)
```

Comment:
I don't know if we use these OSS `deployment` targets for anything - have you tried testing w/ the internal targets as well?

Reply:
Not yet. There are a few differences between OCI images and rules_docker images.

The biggest one is that with the current patch, the OCI images are not pushed by rules_k8s while the rules_docker ones are. 

I was planning to resolve that in a separate PR, but with recent findings in rules_docker, I think we can live without rules_oci for a tiny bit longer.

Let me reduce the scope of this PR to only upgrade rules_k8s to the latest commit.
Once that's done, I will prioritize patches that enable bzlmod support.
And revisit rules_k8s + rules_oci after that.